src/main/groovy/com/github/roroche/plantuml/urls/CompileClasspathUrlsFactory.groovy

```diff
@@ -0,0 +1,38 @@
+package com.github.roroche.plantuml.urls
+
+import org.gradle.api.Project
+
+import static com.github.roroche.plantuml.android.AndroidProjectType.*
+
+/**
+ * Factory for creating the correct classpath based on the type of a project
+ */
+class CompileClasspathUrlsFactory {
```

Comment:
Pretty much the same comment, for these reasons:
- https://www.yegor256.com/2015/02/20/utility-classes-vs-functional-programming.html
- https://www.yegor256.com/2017/11/14/static-factory-methods.html

So you can move the logic to the `SupportedProject` suggested above.